client/src/partials/templates/bhCashBoxFinder.tmpl.html

```diff
@@ -0,0 +1,39 @@
+<div ng-switch="$ctrl.session.state">
+  <div ng-switch-when="finding" class="input-group">
+
+    <div class="input-group-btn" uib-dropdown>
+      <button class="btn btn-default" uib-dropdown-toggle data-cash-box-finder>
+        {{$ctrl.selectedType.label}} <span class="caret"></span>
+      </button>
+      <ul uib-dropdown-menu>
+        <li ng-repeat="type in $ctrl.types">
+          <a href="" ng-click="$ctrl.setType(type)" data-cash-box-finder-option="{{ type.label }}">
+            {{type.label}}
+          </a>
+        </li>
+      </ul>
+    </div>
+
+    <input 
+      bh-cash-box-finder      
+      name="cashBoxFinderValue"
+      ng-model="$ctrl.value"    
+      type="text"
+      class="form-control"
+      uib-typeahead="cashBox as cashBox.text for cashBox in $ctrl.search($viewValue)"
```

Comment:
Why don't you make the `search()` function send an `$http` request and do the filtering in the database?  That way, you do not need to use a regex on the client side.